pure_protobuf/dataclasses_.py

```diff
@@ -117,6 +108,21 @@ def optional_field(number: int, *args, **kwargs) -> Any:
     return field(number, *args, default=None, **kwargs)
 
 
+def one_of(**parts: Tuple[type, int]) -> OneOf_:
+    scheme_ = tuple(
+        OneOfPartInfo(name, type_, number)
+        for name, (type_, number) in parts.items()
+    )
+
+    default = OneOf_(scheme_)
+    return _field(-1, isoneof=True, default=default)
+
+
+def part(type_: Type, number: int) -> Tuple[type, int]:
+    # well, yeah
```

Comment:
Why? 😄

Reply:
Mostly because why not :)

Tuples are small and immutable. They are enough for declaring such a message. 

And `part` is just a good constructor name for user to understand what's going on. Ofc it can be simplified to just tuple passing, but should we?

Reply:
I'd just use a tuple, yet not strictly against the function.

But do you only use it in the tests? Because then I'd prefer to have it in `tests`

Reply:
> But do you only use it in the tests? 

I actually thought of it as of part of public API.

> yet not strictly against the function.

Ok, let's leave it as is for now.

Probably function has advantage in such case:

I changed signature a bit for case when `packed` flag needed:
```python
def part(type_: Type, number: int, *, packed: bool = False) -> Tuple[type, int, bool]:
    ...
```
so user must pass `packed` as named arg:
```python
@message
class A:
    a: OneOf_ = one_of(
        foo=part(int, 1, packed=True)
    )
```

But I do not insist here, it can be changed in every possible way I think